ambassador/ambassador/mapping.py

```diff
@@ -166,16 +206,9 @@ def new_route(self, svc, cluster_name):
                 route['request_headers_to_add'].append({"key": key, "value": value})
 
         cors = self.get('cors')
-        if cors:
-            route['cors'] = {key: value for key, value in {
-                "allow_origin": cors['origins'].split(',') if cors['origins'] else None,
-                "allow_methods": cors.get('methods'),
-                "allow_headers": cors.get('headers'),
-                "expose_headers": cors.get('exposed_headers'),
-                "allow_credentials": cors.get('credentials'),
-                "max_age": cors.get('max_age'),
-                "enabled": True
-            }.items() if value}
+        if cors is not None:
+            route['cors'] = dict()
+            self.generate_route_cors(cors, route['cors'])
```

Comment:
This should probably look like

```python
route['cors`] = self.generate_route_cors(cors)
```

instead of using the `route['cors']` dict as an output parameter.

Reply:
fixed